### PR TITLE
[Misc] Add unit test for _fwd_kernel_ep_scatter_1 and _fwd_kernel_ep_…

### PR DESCRIPTION
**Pull request by pmanczak**
## Purpose

Adds a unit test for the `_fwd_kernel_ep_scatter_1` and `_fwd_kernel_ep_scatter_2`
Triton kernels (DeepGemm group) in `tests/kernels/moe/test_ep_scatter.py`.

Both kernels run inside the `ep_scatter` wrapper and had no upstream coverage:
- `_fwd_kernel_ep_scatter_1` computes per-expert region offsets and fills
  `m_indices` (expert label per output row, -1 for padding), counts aligned to
  BLOCK_E=128.
- `_fwd_kernel_ep_scatter_2` copies each token's activation and per-group scale
  into the row of its (optionally remapped) expert, recording the destination in
  `output_index`.

Phase 1 is checked by comparing `m_indices` against a reference region layout.
Phase 2 is checked by reading the destination back out of the kernel: each token's
`output_index` must point to a row labelled with its expert in `m_indices`, and
that row must hold both the token's activation and its per-group scale. A
non-trivial `expert_map` case is included. Device-agnostic
(`current_platform.device_type`); runs on CUDA and XPU.

## Test Plan

`python -m pytest tests/kernels/moe/test_ep_scatter.py -v`

## Test Result

All 15 cases pass.

### CODE REVIEW

File: tests/kernels/moe/test_ep_scatter.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    recv_topk = torch.zeros(total_tokens, 1, dtype=torch.int32)
+
+    m_indices = _run_scatter(num_experts, recv_topk, tokens_per_expert)[0]
+    ref_m = _ref_m_indices(tokens_per_expert)
+
+    torch.testing.assert_close(m_indices[: ref_m.shape[0]], ref_m, atol=0, rtol=0)
+
+
+# (num_experts, total_tokens, topk, hidden)
+SCATTER_CASES = [
```
Comment (oonyshch) [inline tests/kernels/moe/test_ep_scatter.py:196]: these tests only run `pack_ue8m0=False` / `default block_size=128`, so the int32 UE8M0 packing branch in `_fwd_kernel_ep_scatter_2`  from `deep_gemm_utils.py:199-232` is never tested, imo valid path to add: a case that passes `pack_ue8m0=True` with a uint8 scale, plus `block_size=32` (MXFP8's 32-element scale group) so the non-128 group width is covered too
tho fp8 is not supported on the XPU you are currently testing for... but there exists an XPU (GPU) that supports it ;)

**Comment by oonyshch** [comment]
pls ruff check, ruff format
